toolshed/src/thegraph/deployment_id.rs

```diff
@@ -0,0 +1,254 @@
+use alloy_primitives::B256;
+use async_graphql::Scalar;
+use serde_with::{DeserializeFromStr, SerializeDisplay};
+
+/// A Subgraph's Deployment ID represents unique identifier for a deployed subgraph on The Graph.
+/// This is the content ID of the subgraph's manifest.
+#[derive(
+    Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, SerializeDisplay, DeserializeFromStr,
+)]
+pub struct DeploymentId(pub B256);
```

Comment:
I kept the visibility of the internal data public to avoid breaking changes. But, in my opinion, the internal representation should be opaque, i.e., non-public.